libkbfs/folder_branch_ops.go

```diff
@@ -3960,6 +3950,36 @@ func (fbo *folderBranchOps) notifyOneOpLocked(ctx context.Context,
 	afterUpdateFn func() error) error {
 	fbo.headLock.AssertLocked(lState)
 
+	// We need to get unlinkPath before calling UpdatePointers so that
+	// nodeCache.Unlink can properly update cachedPath.
+	unlinkPath, toUnlink := func() (path, bool) {
+		var (
+			node      Node
+			childName string
+		)
+
+		switch realOp := op.(type) {
+		case *rmOp:
+			node = fbo.nodeCache.Get(realOp.Dir.Unref.Ref())
+			childName = realOp.OldName
+		case *renameOp:
+			node = fbo.nodeCache.Get(realOp.NewDir.Unref.Ref())
```

Comment:
When a file is renamed within the same directory, `realOp.NewDir` is set to the empty value.  In that case, you need to use `realOp.OldDir` here.  See the existing rename code you're (partially) removing.